webapp/src/containers/SwapPage/service.tsx

```diff
@@ -81,7 +81,7 @@ export const handlePoolSwap = async (formState) => {
   let accountToAccountAmount = new BigNumber(0);
 
   // convert account to account, if don't have sufficient funds in one account
-  if (Number(formState.amount1) > maxAmount1) {
+  if (new BigNumber(formState.amount1).gt(maxAmount1)) {
```

Comment:
See line 40 of this same file: `Number(formState.amount1)`, these are the important ones that we need to convert to BigNumber, the values that are passed to the RPC calls, also change all the types for typescript in the RPC call to accept big number. Do this first so that the typescript compiler can tell you where the other parts need to be changed.

Reply:
Will do, Thanks